backend-ci/tests/Services/ProductVariantServiceTest.php

```
@@ -55,6 +55,28 @@ public function test_create_variant_respects_route_product_id(): void
+        $productId = $this->seedProduct(['code' => 'CP100', 'name' => 'Product']);
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('SKU đã tồn tại trong danh sách sản phẩm');
+
+        $this->service->create($productId, ['sku' => 'CP100']);
+    }
+
+    public function test_update_variant_blocks_duplicate_sku(): void
+    {
+        $productId = $this->seedProduct(['code' => 'CP200', 'name' => 'Product']);
+        $firstVariantId = $this->seedVariant($productId, 'SKU-ONE');
```

PHPMD (2.15.0)

Avoid unused local variables such as '$firstVariantId'. (undefined)

(UnusedLocalVariable)